File: fib.lua
#!/usr/bin/luajit

local function fibonacci(n)
    if n<=1 then
        return n
    end
    return fibonacci(n-1) + fibonacci(n-2)
end

io.write("Fibonacci sequence:")
for i = 0, 31 do
    io.write(" " .. fibonacci(i))
end
io.write("\n\n")

File: fib.c
#!/usr/bin/tcc -run
#include <stdio.h>

unsigned int fibbonacci(unsigned int n) {
    if (n <= 1){
        return n;
    }
    return (fibbonacci(n-1) + fibbonacci(n-2));
}

void main() {
    unsigned int i;

    printf("Fibonacci sequence:");
    for(i = 0; i < 32; i++) {
        printf(" %d", fibbonacci(i));            
    }
    printf("\n\n");
}